File: app/src/main/java/mishka/mingeo/di/component/PumpingListActivityComponent.java
package mishka.mingeo.di.component;

import dagger.Component;
import mishka.mingeo.di.ActivityScope;
import mishka.mingeo.di.module.PumpingListModule;
import mishka.mingeo.view.pumpinglist.PumpingListActivity;

@Component(dependencies = {ApplicationComponent.class, ActivityComponent.class}, modules = PumpingListModule.class)
public interface PumpingListActivityComponent {
    void inject(PumpingListActivity activity);
}

File: app/src/main/java/mishka/mingeo/di/component/PumpingListActivityComponent.kt
package mishka.mingeo.di.component

import dagger.Component
import mishka.mingeo.di.ActivityScope
import mishka.mingeo.di.module.PumpingListModule
import mishka.mingeo.view.pumpinglist.PumpingListActivity

@Component(dependencies = [ActivityComponent::class], modules = [PumpingListModule::class])
@ActivityScope
interface PumpingListActivityComponent {
    fun inject(activity: PumpingListActivity)
}